examples/Cassie/cassie_rbt_state_estimator.cc

```diff
@@ -1,140 +1,981 @@
+  output->SetVelocityAtIndex(velocity_index_map_.at("knee_rightdot"),
+                             cassie_out.rightLeg.kneeDrive.velocity);
+  output->SetVelocityAtIndex(velocity_index_map_.at("toe_rightdot"),
+                             cassie_out.rightLeg.footDrive.velocity);
+  output->SetVelocityAtIndex(velocity_index_map_.at("knee_joint_rightdot"),
+                             cassie_out.rightLeg.shinJoint.velocity);
+  output->SetVelocityAtIndex(velocity_index_map_.at("ankle_joint_rightdot"),
+                             cassie_out.rightLeg.tarsusJoint.velocity);
+  output->SetVelocityAtIndex(velocity_index_map_.at("ankle_spring_joint_rightdot"),
+                             0.0);
+
+  // Solve fourbar linkage for heel spring positions
+  double left_heel_spring = 0;
+  double right_heel_spring = 0;
+  VectorXd q = output->GetMutablePositions();
+  if (is_floating_base_) {
+    // Floating-base state doesn't affect the spring values
+    // We assign the floating base of q in case output's floating base is
+    // not initialized.
+    for (int i = 0; i < 7; i++) {
+      q[i] = 0;
+    }
+    q[3] =  1;
+  }
+  solveFourbarLinkage(&left_heel_spring, &right_heel_spring, q);
+  output->SetPositionAtIndex(position_index_map_.at("ankle_spring_joint_left"),
+                             left_heel_spring);
+  output->SetPositionAtIndex(position_index_map_.at("ankle_spring_joint_right"),
+                             right_heel_spring);
+}
+
+
+void CassieRbtStateEstimator::AssignFloatingBaseStateToOutputVector(
+  OutputVector<double>* output, const VectorXd& est_fb_state) const {
+  // TODO(yminchen): Joints names need to be changed when we move to MBP
+  output->SetPositionAtIndex(position_index_map_.at("base_x"), est_fb_state(0));
+  output->SetPositionAtIndex(position_index_map_.at("base_y"), est_fb_state(1));
+  output->SetPositionAtIndex(position_index_map_.at("base_z"), est_fb_state(2));
+  output->SetPositionAtIndex(position_index_map_.at("base_qw"), est_fb_state(3));
+  output->SetPositionAtIndex(position_index_map_.at("base_qx"), est_fb_state(4));
+  output->SetPositionAtIndex(position_index_map_.at("base_qy"), est_fb_state(5));
+  output->SetPositionAtIndex(position_index_map_.at("base_qz"), est_fb_state(6));
+
+  output->SetVelocityAtIndex(velocity_index_map_.at("base_wx"), est_fb_state(7));
+  output->SetVelocityAtIndex(velocity_index_map_.at("base_wy"), est_fb_state(8));
+  output->SetVelocityAtIndex(velocity_index_map_.at("base_wz"), est_fb_state(9));
+  output->SetVelocityAtIndex(velocity_index_map_.at("base_vx"), est_fb_state(10));
+  output->SetVelocityAtIndex(velocity_index_map_.at("base_vy"), est_fb_state(11));
+  output->SetVelocityAtIndex(velocity_index_map_.at("base_vz"), est_fb_state(12));
+}
+
+
+EventStatus CassieRbtStateEstimator::Update(const Context<double>& context,
+    DiscreteValues<double>* discrete_state) const {
+  // Testing
+  /*const auto& cassie_out =
```

Comment:
Remove dead code

Reply:
Is it ok that I leave the codes here because I'm going to use it to fix the "one time-step delay" problem #63.
I added `TODO(yminchen)` to all the codes that should be removed at the end, so I can delete them in the future.